flink-connectors/flink-connector-jdbc/src/test/java/org/apache/flink/connector/jdbc/table/JdbcDynamicTableSinkITCase.java

```diff
@@ -311,6 +311,48 @@ public void testAppend() throws Exception {
                 new String[] {"id", "num", "ts"});
     }
 
+    @Test
+    public void testSqlHintWithTargetColumns() throws Exception {
```

Comment:
I think we can refer to the #18967 and plan/IT tests. The modification is mainly about the planner and we don't need to add tests in the connector package.